connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -157,24 +167,94 @@ public void stop() {
         AppInfoParser.unregisterAppInfo(JMX_PREFIX, workerId);
     }
 
+    public static class MetricGroupId {
+        private final String groupName;
+        private final Map<String, String> tags;
+        private final int hc;
+        private final String str;
+
+        public MetricGroupId(String groupName, Map<String, String> tags) {
+            assert groupName != null;
+            assert tags != null;
+            this.groupName = groupName;
+            this.tags = Collections.unmodifiableMap(new HashMap<>(tags));
```

Comment:
I think this should ensure it preserves the order of the tags.

Reply:
Ack.